src/test/unit_tests/jest_tests/test_versioning_concurrency.test.js

```diff
@@ -81,4 +77,68 @@ describe('test versioning concurrency', () => {
         await P.delay(1000);
         expect(number_of_successful_operations.length).toBe(15);
     });
+
+    it('concurrent delete of latest version', async () => {
+        const bucket = 'bucket1';
+        const key = 'key3';
+        const number_of_versions = 5;
+        const versions_arr = await _upload_versions(bucket, key, number_of_versions);
+        expect(versions_arr.length).toBe(number_of_versions);
+
+        const successful_operations = [];
+        for (let i = 0; i < 3; i++) {
```

Comment:
maybe worth increasing iteration to increase risk for concurrency issue

Reply:
@shirady as long as you see a retry every time I think it's ok